pkg/app/pipedv1/plugin/kubernetes/deployment/plugin.go

```diff
@@ -141,31 +139,29 @@ func (p *Plugin) DetermineVersions(ctx context.Context, _ *sdk.ConfigNone, input
 }
 
 // DetermineStrategy determines the strategy for the deployment.
-func (p *Plugin) DetermineStrategy(ctx context.Context, _ *sdk.ConfigNone, input *sdk.DetermineStrategyInput) (*sdk.DetermineStrategyResponse, error) {
+func (p *Plugin) DetermineStrategy(ctx context.Context, _ *sdk.ConfigNone, input *sdk.DetermineStrategyInput[kubeconfig.KubernetesApplicationSpec]) (*sdk.DetermineStrategyResponse, error) {
 	logger := input.Logger
 	loader := provider.NewLoader(toolregistry.NewRegistry(input.Client.ToolRegistry()))
 
-	cfg, err := config.DecodeYAML[*kubeconfig.KubernetesApplicationSpec](input.Request.TargetDeploymentSource.ApplicationConfig)
-	if err != nil {
-		logger.Error("Failed while decoding application config", zap.Error(err))
-		return nil, err
+	cfg := input.Request.TargetDeploymentSource.ApplicationConfig.Spec
```

Comment:
[imo] It would be nice to add a method into DeploySource to get ApplicationConfig from DeploySource, including a nil check because `ApplicationConfig` is a pointer.
WDYT?

Reply:
I agree with you and I added AppConfig method
https://github.com/pipe-cd/pipecd/pull/5740/commits/53ef85092c50be8c47ceeb5a3b03af9e2345b68b